src/App.tsx

```diff
@@ -1,26 +1,149 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import cn from 'classnames';
+
+import { Todo } from './types/Todo';
+import { ErrorType } from './types/ErrorType';
+import { FilterType } from './types/FilterType';
+
 import { UserWarning } from './UserWarning';
+import { getTodos, USER_ID } from './api/todos';
+
+import { handleDeleteTodo } from './utils/handleDeleteTodo';
+import { getFilteredTodos } from './utils/getFilteredTodo';
 
-const USER_ID = 0;
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { Header } from './components/Header';
+import { TodoItem } from './components/TodoItem';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [deletingTodoIds, setDeletingTodoIds] = useState<number[]>([]);
+  const [filter, setFilter] = useState(FilterType.all);
+  const [errorMessage, setErrorMessage] = useState<ErrorType>(
+    ErrorType.default,
+  );
+
+  const handleFilterChange = useCallback(
+    (filterBy: FilterType) => {
+      setFilter(filterBy);
+    },
+    [setFilter],
+  );
+
+  const filteredTodos = useMemo(
+    () => getFilteredTodos(todos, filter),
+    [todos, filter],
+  );
+
+  const timeoutRef = useRef<NodeJS.Timeout | null>(null);
+
+  const handleRemoveError = useCallback(() => {
+    setErrorMessage(ErrorType.default);
+  }, []);
+
+  const handleErrorMessage = useCallback(
+    (error: ErrorType) => {
+      setErrorMessage(error);
+
+      if (timeoutRef.current) {
+        clearTimeout(timeoutRef.current);
+      }
+
+      timeoutRef.current = setTimeout(() => {
+        handleRemoveError();
+        timeoutRef.current = null;
+      }, 3000);
+    },
+    [handleRemoveError],
+  );
+
+  const handleTempTodo = useCallback(
+    (todo: Todo | null) => {
+      setTempTodo(todo);
+    },
+    [setTempTodo],
+  );
+
+  useEffect(() => {
+    handleDeleteTodo(
+      deletingTodoIds,
+      setTodos,
+      setDeletingTodoIds,
+      handleErrorMessage,
+    );
+  }, [deletingTodoIds, handleErrorMessage]);
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => handleErrorMessage(ErrorType.load));
```

Comment:
It's better to move the logic inside the `useEffect `hook to a separate function, as it improves code readability, reusability, and makes testing easier. It also helps keep `useEffect `focused on side effects.